### ISSUE TITLE
Error during "cds build --production" in cds-typer/cds-plugin.js:139:19

### ISSUE THREAD

**Issue by D027152** [cap-js/cds-typer#366]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Nature of Your Project

TypeScript

### Current Behavior

We have added typescript to our project (cds add typescript). In our pipeline, when execution cds build --production, we get an error in @cap-js/cds-typer/cds-plugin.js:139:19, see for example this log:
https://github.tools.sap/erp4sme/crypto-for-business/actions/runs/8655257/job/34323380?pr=4253

For me it looks as if the typescript compiler does not know the command.

Locally we don't have this problem. 

### Expected Behavior

cds build --production runs without error

### Steps To Reproduce

repo https://github.tools.sap/erp4sme/crypto-for-business/
branch new_cds-build-error contains the state that creates the error in the pipeline

### Environment

```markdown
- **OS**:
- **Node**: v18.18.0   
- **npm**: 9.8.1
- **cds-typer**: 0.27.0
- **cds**: 8.3.1
```

### Repository Containing a Minimal Reproducible Example

https://github.tools.sap/erp4sme/crypto-for-business/tree/new_cds-build-error

**Comment by Markus-Schroeder-de** [comment]
sorry, can't access your repo, but it looks liike that i have the same Problem... Building the project works perfect, but if i use cds build, i get also an error in the same file and line.  The cds build didn't find the modules.

**Comment by Markus-Schroeder-de** [comment]
have a look in the file "cds-plugin.js". They check there if a "tsconfig.cdsbuild.json" exists. We have different tsconfig-files in our project, because while it's building we need different path to the /gen/ folder. after i created this file in the root-Dir, i get different error. so i assume, you need also a seperat tsconfig-file for your project. i will go on with the investigation, but maybe this helps a little bit.

**Comment by Markus-Schroeder-de** [comment]
have a look at https://cap.cloud.sap/docs/tools/cds-typer#integrate-into-your-ci

**Comment by daogrady** [comment]
Hi everyone,

thanks for reporting this issue. I inspected your project locally and was unable to reproduce the exact error you were facing. Could you please reproduce the problem, but set `DEBUG=build` so we can gather further insights?

@Markus-Schroeder-de 
> They check there if a "tsconfig.cdsbuild.json" exists

That is partially correct. The build task uses any _tsconfig.json_ that is found in the root directory of the project, assuming that this is the effective tsconfig the project is supposed to be started and built with. This config is mainly needed to build your project exactly how you would locally, and to determine the directory the generated types are put in (`compilerOptions.paths.#cds-models...`).

As users, as in your case, may use different configs for development and production, we offer a breakout from this assumption. Users can put a _tsconfig.cdsbuild.json_ in the project's root directory to customise the build parameters used for `tsc` however they want for the TypeScript build step.

tl;dr: _tsconfig.cdsbuild.json_ is only required if you (a) don't have a _tsconfig.json_ in your root directory or (b) want to use different parameters during build than what you have in your _tsconfig.json_.

If the error you are encountering differs from what @D027152 found, please share the error message and preferably a minimal reproduction example.

Best,
Daniel

**Comment by D027152** [comment]
Hi @daogrady and @Markus-Schroeder-de 

The hint with the tsconfig file was actually good. I saw that in our dockerfile we include a file tsconfig.deploy.json (not the  tsconfig.json). So the build task that you, @daogrady, mention, did apparently not find any tsconfig file, resulting in the error that I got. 
I have now created a tsconfig.cdsbuild.json and included it in the dockerfile. Now the build works. 
So I don't get any "different error" that you, @Markus-Schroeder-de, mention. 

Thank you,
Stefan

**Comment by daogrady** [comment]
Hi @D027152,

glad you got it working! I'll close this issue as fixed then.

@Markus-Schroeder-de I assume your problem lies in the same domain. Please make sure there is a _tsconfig.json_ available in your project root when you are building the project, or supply a _tsconfig.cdsbuild.json_ instead if your are using another config file during build time to point the build task in the right direction.

Assuming you have a config file specific for building/ deploying that is called _tsconfig.deploy.json_, you could create a _tsconfig.cdsbuild.json_ with the content:

```json
{
  "extends": "./tsconfig.deploy.json",
}
```

which will effectively use your _tsconfig.deploy.json_ during build without any modifications.
Feel free to comment here or open another issue if the problem persists.

Best,
Daniel
